docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,421 @@
+| ----------------------------- | ------------------------------ | ------------------- | ------------------------------------------------------------- | ------------------------------- |
+| [React Cache](#react-cache)   | Return values of functions     | Server              | Reduce data store calls and share data between route segments | Per-request lifecycle           |
+| [Data Cache](#data-cache)     | Return values of data requests | Server              | Store data across user requests and deployments               | Persistent (can be revalidated) |
+| [Render Cache](#render-cache) | Rendered React Components      | Server              | Reduce rendering cost and improve performance                 | Persistent (can be revalidated) |
+| [Router Cache](#router-cache) | Route Segments (RSC Payload)   | Client              | Reduce server requests and improve navigation experience      | User session or time-based.     |
+
+## How caching works
+
+### React Cache
+
+React's `cache` function allows you to **memoize** the return value of a function, saving the result to memory. This means you can call the same function multiple times while only executing it once during a render pass.
+
+In Next.js, the [`fetch` API](#fetch) uses `cache` to **deduplicate** requests for the same data. Similar to memoization, request deduplication is an optimization that takes **multiple** requests for the same data and only makes a **single** request to the data source. This is done by storing the result of the first request in a temporary server-side cache, and returning the cached result for subsequent requests.
+
+{/* Diagram */}
+
+Since `fetch` requests are automatically deduplicated, you can fetch data in the components that need it without props drilling and worrying about the performance impact of making multiple requests for the same data. Deduplication applies to `fetch` requests in `generateMetadata`, `generateStaticParams`, Layouts, Pages, and other Server Components.
+
+For cases where `fetch` is not suitable (e.g. database clients, CMS clients, or GraphQL), you can use React's `cache` function to manually deduplicate requests. Although Next.js uses `cache` to memoize data requests on the server, the `cache` function can also be used to memoize any function on the server or the client.
+
+> **Memoization vs. Deduplication**:
+>
+> You may see the terms **memoization** and **deduplication** being used interchangeably. Memoization is a general term that applies storing the return values of **any function** in-memory, whereas request deduplication applies specifically to **data requests**.
+
+#### Duration
+
+The cache lasts the lifetime of a server request until the rendering pass completes.
+
+#### Revalidating
+
+Since the cache is cleared after each request, there is no need to revalidate the cache.
```

Comment:
```suggestion
React's `cache` function is specifically for memoizing the return values of functions when rendering the React tree. Since the memoization only applies to rendering the React tree and is not shared across requests there is no revalidation. 
```